airbyte-integrations/connectors/source-chargebee/source_chargebee/source.py

```diff
@@ -0,0 +1,142 @@
+    InvalidRequestError,
+    OperationFailedError,
+)
+
+# Airbyte
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.http import HttpStream
+from airbyte_cdk.sources.streams.http.auth import TokenAuthenticator
+from airbyte_cdk.models import AirbyteStream, SyncMode
+
+
+class ChargebeeStream(Stream):
+    # No support for incremetal sync across Chargebee endpoints
+    supports_incremental = True
+    primary_key = "id"
+
+    def __init__(self, env, limit):
+        self.env = env
+        self.limit = limit
+        super().__init__()    
+
+    def read_records(
+        self,
+        sync_mode: SyncMode,
+        cursor_field: List[str] = None,
+        stream_slice: Mapping[str, Any] = None,
+        stream_state: Mapping[str, Any] = None,
+    ) -> Iterable[Mapping[str, Any]]:
+
+        result = self.api.list(
+            {
+                "env": self.env,
+                "limit": self.limit,
+            }
+        )
+        for message in result:
+            yield message._response
+
+    @abstractmethod
+    def cursor_field(self) -> str:
+        """
+        Defining a cursor field indicates that a stream is incremental, 
+        so any incremental stream must extend this class
+        and define a cursor field.
+        """
+        pass
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
+        """
+        Return the latest state by comparing the cursor value 
+        in the latest record with the stream's most recent state object
+        and returning an updated state object.
+        """
+        return {
+            self.cursor_field: max(
+                latest_record.get(self.cursor_field), 
+                current_stream_state.get(self.cursor_field, 0))
+        }
+
+
+class SubscriptionStream(ChargebeeStream):
+    name = "Subscription"
+    cursor_field = "updated_at"
+    api = Subscription
+
+
+class CustomerStream(ChargebeeStream):
+    name = "Customer"
+    cursor_field = "updated_at"
+    api = Customer
+
+
+class SourceChargebee(AbstractSource):
+    LIMIT = 100
+
+    def __init__(self):
+        self.env = None
+        super().__init__()
+
+    def _get_client(self, config: Mapping, logger):
+        """Construct client"""
+        # logger.info(f"GET CLIENT: Before configuration=")
+        # configuration = {"api_key": config["api_key"], "site": config["site"]}
+        logger.info(f"GET CLIENT: Before client=")
+        client = ChargeBee
+        logger.info(f"GET CLIENT: Before client.configure=")
+        client.configure(options=config)
+        logger.info(f"GET CLIENT: Before self.env=")
+        self.env = client.default_env
+        logger.info(f"GET CLIENT: End")
+
+    def check_connection(self, logger, config: Mapping[str, Any]) -> Tuple[bool, any]:
+        logger.info(f"STARTING CHECK CONNECTION") # Debugging
```

Comment:
there seems to be a bug with the chargebee python client. Instead you can make this work by using the vanilla `requests` library like so: 

```
            value = 'Basic ' + base64.b64encode(('%s:' % config['api_key']).encode('latin1')).strip().decode('latin1')
            response = requests.get('https://kabo-test.chargebee.com/api/v2/subscriptions', headers={'Authorization': value})
            response.raise_for_status()
```

Reply:
Hi @sherifnada @davinchia, I actually was curious as to why Airbyte couldn't find "api_key" from the config.

So, from `spec.json`, I removed `"airbyte_secret": true`, which solved the problem.
Is this because of how Airbyte is the encoding & passing secrets?

Reply:
`spec.json` is used by the UI to create the configuration form that the user fills out, and the JSON object which eventually is passed into the connector as `config`. This `spec.json` currently declares inputs `site_api_key` and `site`. Which means that when the user fills it out in the UI, Airbyte will send the connector a config which looks like 
```
{ site_api_key: xyz, site: abc}
```

However the Chargebee API expects a parameter called `api_key`. 

This is confusing because when running locally you _can_ call the connector with a config that doesn't match `spec.json` e.g: `{api_key: xyz, site: abc}` (input validation will be implemented in #2711 ). My guess is your local config is correct but because the spec is incorrect the UI generates the wrong object. `airbyte_secret:true` should be unrelated.